app/src/main/java/com/mapbox/navigation/examples/standalone/background/BackgroundExampleActivity.kt

```diff
@@ -0,0 +1,474 @@
+        // that make sure the route request is optimized
+        // to allow for support of all of the Navigation SDK features
+        mapboxNavigation.requestRoutes(
+            RouteOptions.builder()
+                .applyDefaultNavigationOptions()
+                .applyLanguageAndVoiceUnitOptions(this)
+                .coordinatesList(listOf(originPoint, destination))
+                // provide the bearing for the origin of the request to ensure
+                // that the returned route faces in the direction of the current user movement
+                .bearingsList(
+                    listOf(
+                        Bearing.builder()
+                            .angle(originLocation.bearing.toDouble())
+                            .degrees(45.0)
+                            .build(),
+                        null
+                    )
+                )
+                .layersList(listOf(mapboxNavigation.getZLevel(), null))
+                .build(),
+            object : NavigationRouterCallback {
+                override fun onCanceled(routeOptions: RouteOptions, routerOrigin: RouterOrigin) {
+                    // no impl
+                }
+
+                override fun onFailure(reasons: List<RouterFailure>, routeOptions: RouteOptions) {
+                    // no impl
+                }
+
+                override fun onRoutesReady(
+                    routes: List<NavigationRoute>,
+                    routerOrigin: RouterOrigin
+                ) {
+                    setRouteAndStartNavigation(routes)
+                }
+            }
+        )
+    }
+
+    private fun setRouteAndStartNavigation(routes: List<NavigationRoute>) {
+        // set routes, where the first route in the list is the primary route that
+        // will be used for active guidance
+        mapboxNavigation.setNavigationRoutes(routes)
+        navigationCamera.requestNavigationCameraToFollowing()
+        // start background service which won't let `MapboxNavigation` to be destroyed
+        startService(Intent(this, BackgroundService::class.java))
+    }
+
+    private fun clearRouteAndStopNavigation() {
+        // clear
+        mapboxNavigation.setNavigationRoutes(listOf())
+        navigationCamera.requestNavigationCameraToOverview()
+        // Stop background service. `MapboxNavigation` will be destroyed as soon as user leaves
+        // current activity
+        stopService(Intent(this, BackgroundService::class.java))
+    }
+}
+
+/***
+ * The idea of this service is to let `MapboxNavigationApp` know that MapboxNavigation is in use
+ * and should not be destroyed. While active guidance session is running, MapboxNavigation keeps
+ * its own foreground service running.
+ * You can disable built-in foreground service by calling `MapboxNavigation.startTripSession(false)`
+ * and implement notification on your own.
+ */
+class BackgroundService : Service(), LifecycleOwner {
```

Comment:
How is custom background service better than our own foreground service?
Also, how is MapboxNavigationApp going to know about this service? I don't see it being notified.

Reply:
> How is custom background service better than our own foreground service?

Built-in navigation foreground service doesn't prevent `MapboxNavigationApp` from being destroyed when user leaves activity.

> Also, how is MapboxNavigationApp going to know about this service? I don't see it being notified.

Under the hood, `requireMapboxNavigation` creates `MapboxNavigation` via `MapboxNavigationApp` using current object lifecycle

Reply:
Does it need to be a foreground service? Since Nav SDK already has one, I think this can be a simple `object`, which implements `LifecycleOwner` and has `start` and `stop` functions.

Reply:
> Under the hood, requireMapboxNavigation creates MapboxNavigation via MapboxNavigationApp using current object lifecycle

Ah, the important part here is that inside this service we create mapboxNavigation once again, right? We still use the old instance, but by creating it here we attach this service, and `MapboxNavigationApp` does not destroy `MapboxNavigation` while the service is alive?